tests/test_cratedb_polars_read.py

```diff
@@ -0,0 +1,43 @@
+import sys
+
+import polars as pl
+import pytest
+import sqlalchemy as sa
+from polars.testing import assert_frame_equal
+
+REFERENCE_FRAME = pl.from_records([{"mountain": "Mont Blanc", "height": 4808}])
+SQL_SELECT_STATEMENT = "SELECT mountain, height FROM sys.summits ORDER BY height DESC LIMIT 1;"
+
+
+if sys.version_info < (3, 8):
+    pytest.skip("Does not work on Python 3.7", allow_module_level=True)
+
+
+def test_crate_read_sql(cratedb_http_host, cratedb_http_port):
+    engine = sa.create_engine(
+        url=f"crate://{cratedb_http_host}:{cratedb_http_port}",
+        echo=True,
+    )
+    conn = engine.connect()
+    df = pl.read_database(
+        query=SQL_SELECT_STATEMENT,
+        connection=conn,
+        schema_overrides={"value": pl.Utf8},
+    )
+    assert_frame_equal(df, REFERENCE_FRAME)
+
+
+def test_psycopg_read_sql(cratedb_psql_host, cratedb_psql_port):
+    engine = sa.create_engine(
+        url=f"postgresql+psycopg_relaxed://crate@{cratedb_psql_host}:{cratedb_psql_port}",
+        isolation_level="AUTOCOMMIT",
+        use_native_hstore=False,
+        echo=True,
+    )
+    conn = engine.connect()
+    df = pl.read_database(
+        query=SQL_SELECT_STATEMENT,
+        connection=conn,
+        schema_overrides={"value": pl.Utf8},
+    )
+    assert_frame_equal(df, REFERENCE_FRAME)
```

Comment:
Dear @surister,

I am only now reading your comment over at https://github.com/crate/crate/issues/12952#issuecomment-1862944511. Can I humbly ask you to check whether the operation on 600K rows worth of data is faster using the Psycopg3 driver, compared to the HTTP/JSON-based driver from [crate-python](https://pypi.org/project/crate/)?


```console
pip install 'polars[pyarrow]<0.21' sqlalchemy-postgresql-relaxed
```
```python
import polars as pl
import sqlalchemy as sa

engine = sa.create_engine("crate+psycopg://crate@localhost/doc")

df = pl.read_database(
    query="SELECT * FROM metrics",
    connection=engine,
    schema_overrides={"value": pl.Utf8},
)
print(df)
```

With kind regards,
Andreas.

NB: This will not run the CrateDB SQLAlchemy dialect though, so it will not support any special data types. There is another patch which will bring everything together correspondingly.

- https://github.com/crate-workbench/sqlalchemy-cratedb/pull/11

Reply:
- read_database - postgresql+psycopg_relaxed  8.5
- read_database_uri - postgresql 5.13
- read_database - crate 8.8s